projects/NXP/devices/iMX6/linux/linux.arm.conf

```diff
@@ -142,7 +142,6 @@ CONFIG_INITRAMFS_FORCE=y
 # CONFIG_RD_XZ is not set
 # CONFIG_RD_LZO is not set
 # CONFIG_RD_LZ4 is not set
-CONFIG_INITRAMFS_COMPRESSION=""
```

Comment:
Keep this.

Reply:
menuconfig removes it

Reply:
`make oldconfig` adds it.

Reply:
http://ix.io/1PJx

Reply:
If you're running `make oldconfig` or `make menuconfig` with `CONFIG_INITRAMFS_SOURCE=""` then `CONFIG_INITRAMFS_COMPRESSION=""` will naturally drop out (as it's dependent on `CONFIG_INITRAMFS_SOURCE` != ""), but that's not how we build so you need to test the config with a dummy path configured for `CONFIG_INITRAMFS_SOURCE` else `CONFIG_INITRAMFS_COMPRESSION` will keep being added back into `.config`

Reply:
> CONFIG_INITRAMFS_COMPRESSION

but why does it matter at all?

Reply:
Really?

Reply:
> Really?

If we require `CONFIG_INITRAMFS_SOURCE=""` to be set so we can use a clean config then all the configs should have a dummy path inserted in so it can be replaced later and the config doesn't change.

Reply:
The work flow should be (as always)
```
cd build.LibreELEC-iMX6.arm-9.80-devel/linux-5.2/
cp ../../projects/NXP/devices/iMX6/linux/linux.arm.conf .config                                                                                
ARCH=arm make menuconfig                                                                                                                       
cp .config ../../projects/NXP/devices/iMX6/linux/linux.arm.conf
```

We should not have to alter the config in any way or have to edit the result of `menuconfig` or `oldconfig`

That is all that I'm trying to point out

Reply:
Sure. Or just run your test with a dummy value and then remove it afterwards. Either way, this config is currently wrong in the sense that the config should not be changed (options added/removed by the kernel) during the build.

Reply:
> Sure. Or just run your test with a dummy value and then remove it afterwards. 

I'm not going to do that. Having to do that will make it very unfriendly to other developers.

Reply:
In order to reliably run `make oldconfig` you must apply the same config changes that [linux/package.mk](https://github.com/LibreELEC/LibreELEC.tv/blob/989f0f3fbf3490bf52d335423d7ca2049151e641/packages/linux/package.mk#L66-L104) is making, which is not in your workflow.

I've often thought we should have a tool to help with running `make oldconfig` because clearly this is something we need to get right.

Reply:
> I'm not going to do that. Having to do that will make it very unfriendly to other developers.

Your workflow is flawed, sorry about that.

Reply:
> > I'm not going to do that. Having to do that will make it very unfriendly to other developers.
> 
> Your workflow is flawed, sorry about that.

https://openelec.tv/documentation/development-workflow#kernel-config

Reply:
Those instructions are wrong - it's fine up until the new config is copied over the `post_patch()`'ed config which thus loses any config changes made specifically for the build system.

Reply:
> Those instructions are wrong - it's fine up until the new config is copied over the `post_patch()`'ed config which thus loses any config changes made specifically for the build system.

It's how it was always intended to be done. The config should not have to be manually edited every time someone changes the config. So either we fix the way the kernel config is edited in post_patch or we fix the kernel configs

Reply:
Sounds good.

Reply:
We don't use the kernel config that lives in the directory after running `./scripts/unpack linux` because it would add all the options that we change and we would have to manually edit the config all the time.

```
diff --git a/projects/NXP/devices/iMX6/linux/linux.arm.conf b/projects/NXP/devices/iMX6/linux/linux.arm.conf
index 064cec1b71..8284ad383a 100644
--- a/projects/NXP/devices/iMX6/linux/linux.arm.conf
+++ b/projects/NXP/devices/iMX6/linux/linux.arm.conf
@@ -32,7 +32,7 @@ CONFIG_HAVE_KERNEL_LZ4=y
 # CONFIG_KERNEL_XZ is not set
 CONFIG_KERNEL_LZO=y
 # CONFIG_KERNEL_LZ4 is not set
-CONFIG_DEFAULT_HOSTNAME="@DISTRONAME@"
+CONFIG_DEFAULT_HOSTNAME="LibreELEC"
 CONFIG_SWAP=y
 CONFIG_SYSVIPC=y
 CONFIG_SYSVIPC_SYSCTL=y
@@ -134,14 +134,17 @@ CONFIG_CGROUPS=y
 # CONFIG_SYSFS_DEPRECATED is not set
 CONFIG_RELAY=y
 CONFIG_BLK_DEV_INITRD=y
-CONFIG_INITRAMFS_SOURCE=""
+CONFIG_INITRAMFS_SOURCE="/home/lukas/libreelec/build.LibreELEC-iMX6.arm-9.80-devel/image/initramfs.cpio"
 CONFIG_INITRAMFS_FORCE=y
+CONFIG_INITRAMFS_ROOT_UID=0
+CONFIG_INITRAMFS_ROOT_GID=0
 # CONFIG_RD_GZIP is not set
 # CONFIG_RD_BZIP2 is not set
 # CONFIG_RD_LZMA is not set
 # CONFIG_RD_XZ is not set
 # CONFIG_RD_LZO is not set
 # CONFIG_RD_LZ4 is not set
+CONFIG_INITRAMFS_COMPRESSION=""
 CONFIG_CC_OPTIMIZE_FOR_PERFORMANCE=y
 # CONFIG_CC_OPTIMIZE_FOR_SIZE is not set
 CONFIG_SYSCTL=y
```

Reply:
Yes, that's true, but the issue is that you can't currently run `make oldconfig` on the "raw" config (from the project folder) as it's not the same config that will be used during the actual build, and consequently we end up with options being added/removed that should not be added or removed.

The current workflow of "copy in a new config and run `make oldconfig`" is a little naiive as it doesn't result in an accurate representation of the config used during the build. Maybe this has only become an issue since we dropped initramfs compression - prior to this change I guess it went unnoticed (or didn't drop out?)

The simplest solution might be to change the kernel configs for all projects by specifying a dummy path for `CONFIG_INITRAMFS_SOURCE="/dummy/path/initramfs.cpio"`, and adding `CONFIG_INITRAMFS_ROOT_UID=0`/`CONFIG_INITRAMFS_ROOT_GID=0` (removing the latter from `linux/package.mk`).